### ISSUE TITLE
None of the exunit tests run

### ISSUE THREAD

**Issue by nickva** [apache/couchdb#4327]
Noticed it locally at first then confirmed in CI.

```

[2022-12-17T05:42:33.927Z] warning: :ssl_verify_hostname.verify_fun/3 defined in application 
 :ssl_verify_fun is used by the current application but the current application does not depend on 
 :ssl_verify_fun. To fix this, you must do one of:
[2022-12-17T05:42:33.927Z] 
[2022-12-17T05:42:33.927Z]   1. If :ssl_verify_fun is part of Erlang/Elixir, you must include it  
 under :extra_applications inside "def application" in your mix.exs
[2022-12-17T05:42:33.927Z] 
[2022-12-17T05:42:33.927Z]   2. If :ssl_verify_fun is a dependency, make sure it is listed under 
 "def deps" in your mix.exs
[2022-12-17T05:42:33.927Z] 
[2022-12-17T05:42:33.927Z]   3. In case you don't want to add a requirement to :ssl_verify_fun, you may optionally skip this warning by adding [xref: [exclude: [:ssl_verify_hostname]]] to your "def project" in mix.exs
[2022-12-17T05:42:33.927Z] 
[2022-12-17T05:42:33.927Z] Invalid call found at 2 locations:
[2022-12-17T05:42:33.927Z]   test/elixir/lib/couch.ex:69: Couch.process_arguments/3
[2022-12-17T05:42:33.927Z]   test/elixir/lib/couch_raw.ex:2: Rawresp.process_arguments/3
[2022-12-17T05:42:33.927Z] 
[2022-12-17T05:42:33.927Z] Generated couchdbtest app
[2022-12-17T05:42:33.928Z] There are no tests to run
```

In the meantime `make exunit` stage seems happy and green

<img width="987" alt="Screen Shot 2022-12-21 at 1 20 02 AM" src="https://user-images.githubusercontent.com/211822/208835494-65509e0a-1cb5-473c-aa6a-097c3f0b6855.png">

**Comment by nickva** [comment]
Tried jumping back to

```
commit 3c24731a5e49bbb4a1d1f407f11f141ca8698e6c (HEAD)
Date:   Wed Nov 16 22:32:35 2022
```

Same results (tests ignored) jumping at least a month back, so not related to recent rebar/mix changes.

**Comment by nickva** [comment]
I guess things are not that dire, as it looks we only have one exunit test suite in dreyfus.

```
find . -name elixir -type d | grep -v '\./test/elixir'
./src/dreyfus/test/elixir
```

The tests are:

https://github.com/apache/couchdb/blob/main/src/dreyfus/test/elixir/test/search_test.exs
https://github.com/apache/couchdb/blob/main/src/dreyfus/test/elixir/test/partition_search_test.exs

Should we rewrite them in Erlang as well?

**Comment by janl** [comment]
there are a few more, but not a lot:

```
> find  . -name exunit 
./src/couch/test/exunit
> ls src/couch/test/exunit/
couch_compress_tests.exs	same_site_cookie_tests.exs	test_helper.exs
```

**Comment by nickva** [comment]
Thanks, @janl. A few more it seems. It might be a small beginner task to switch them to Erlang then we should be able to reduce our makefile and mix.exs size a bit. Also make the test suite run faster a bit.

**Comment by pgj** [comment]
I am interested in working on this task.  Looks like I cannot assign it to myself though -- could somebody help me with this please?

**Comment by nickva** [comment]
@pgj thank you for taking a look at it

**Comment by iilyak** [comment]
I think we should move in another direction. Move eunit tests into elixir. My 2 cents.

**Comment by jaydoane** [comment]
> I think we should move in another direction. Move eunit tests into elixir. My 2 cents.

I think elixir tests were an interesting experiment, but I have found that they add to the friction of developing and maintaining CouchDB.

**Comment by nickva** [comment]
> I think we should move in another direction. Move eunit tests into elixir. My 2 cents.

@iilyak ideally, yes. But in 2-3 years we ended up with only 2 exunit files.  Elixir worked great for integration tests moving from Javascript to Elixir, and we got a lot of help from contributors there, but there just wasn't much interest in translating Erlang eunit tests to Elixir.

In the meantime EUnit tests have improved quite a bit: we have maps, so asserts matching on response shapes is more ergonomic, the TDEF/TDEF_FE macros make test suite nice to look at, we get better coverage reports, which helps immensely identifying areas we completely missed.
